### PR TITLE
Add selecting elements by namespace (#1811)

### PR DESCRIPTION
**Pull request by mo20053444**
This commit enables users to select all elements within a particular namespace by inputting queries like "ns|*".

### CODE REVIEW

File: src/main/java/org/jsoup/select/QueryParser.java [hunk 1]
```diff
@@ -162,6 +162,8 @@ private void findElements() {
             byId();
         else if (tq.matchChomp("."))
             byClass();
+        else if (tq.toString().endsWith("|*"))
```
Comment (jhy) [inline src/main/java/org/jsoup/select/QueryParser.java:165]: This is not the right way to do this. The TokenQueue must be consumed token by token. With this implementation, queries like `ns|* div` (or anything after the namespace wildcard selector) will fail. You will need an appropriate matcher in TokenQueue. I think you could update the current tag matcher.

File: src/main/java/org/jsoup/select/QueryParser.java [hunk 2]
```diff
@@ -262,6 +264,12 @@ private void byTag() {
         }
     }
 
+    private void byNamespace() {
```
Comment (jhy) [inline src/main/java/org/jsoup/select/QueryParser.java:267]: I think I would just extend the byTag method to support this.

File: src/test/java/org/jsoup/nodes/ElementTest.java [hunk 3]
```diff
@@ -1319,6 +1319,16 @@ public void testNamespacedElements() {
         assertEquals("html > body > fb|comments", els.get(0).cssSelector());
     }
 
+    @Test
+    public void testSelectByNamespace() {
```
Comment (jhy) [inline src/test/java/org/jsoup/nodes/ElementTest.java:1323]: This should be in SelectorTest, and needs to be more detailed -- e.g. to catch the example I pointed out, and validate other variations work.

**Review by jhy — CHANGES_REQUESTED** [review]
Review notes inline.

**Comment by mo20053444** [comment]
@jhy Hi Jonathan, hope you had a great holiday season and thank you for your feedback! I will take a look at your comments and make changes accordingly. Thanks!

**Pull request closed without merging**